### DISCUSSION TITLE
VSCode node_modules auto-import intellisense not working?

### DISCUSSION THREAD

**Discussion by Tenrys** [nuxt/nuxt#22284] [Questions]
Hello,

I setup a new project recently, and if I try to type the name of an import from some module, I get no suggestions at all. Taking for example `@vueuse/core`, which I have installed in my project. Am I doing something wrong?

https://github.com/nuxt/nuxt/assets/3979239/25ee3714-6810-4f28-9091-d26442834384

**Comment by Jannchie** [comment]
Do you have `@vueuse/nuxt` installed? If you have it installed and `nuxt.config.ts` is configured like this:

``` ts
export default defineNuxtConfig({
  modules: [
    '@vueuse/nuxt',
  ],
})
```

You should theoretically be able to get auto-import intellisense.

If that doesn't work, you may need to restart vscode, or restart the language server.

**Reply by Tenrys** [reply]
Well, VueUse was an example. I would be having this problem with any other module I want to import, wouldn't I? I will try what you suggested for it though.

**Reply by Jannchie** [reply]
In fact, third-party libraries are not provided with "auto-import" functionality by default... In most cases, you will need to import manually.

**Reply by Tenrys** [reply]
I'm not talking about the feature where you can just type the name of something and start using it without importing it into the file.

I'm talking about VSCode being able to find symbols throughout dependencies and automatically writing the import statement for me so that I can use them directly, without using Nuxt's auto imports. It works just like this in other Vue projects, so I don't see a reason why Nuxt would prevent it from functioning.

**Reply by Jannchie** [reply]
Ah, I misunderstood. But it shouldn't be. Perhaps,  <kbd>cmd</kbd> + <kbd>p</kbd> then typing `> Typescript: Restart TS Server` to restart ts server would help..

**Reply by Tenrys** [reply]
I'm using Volar in takeover mode, so I don't really have that option, but I tried with the builtin TypeScript extension and it didn't work either.

**Reply by AntonioDell** [reply]
@Tenrys did you manage to solve this? Until last week I didn't have any issues with auto-import intellisense, but somehow I get the same behavior you are describing now.
Also using Volar in takeover mode, but not using Nuxt.
Maybe important: I am using a monorepo approach with pnpm workspaces.

Any hint as to what caused this issue or how to solve this is appreciated!

**Reply by Tenrys** [reply]
I don't remember honestly, I put the nuxt project I had on hold. But I had a similar issue with Volar in general in a different project, over [here](https://github.com/vuejs/language-tools/issues/3730).

**Reply by AntonioDell** [reply]
Thank you! Reverting to 1.2.0 did the trick. Maybe the version of Volar got updated in the last week in our project and that's why the issue came up.

**Reply by AntonioDell** [reply]
Figured it out with the help of a colleague. The Problem is the Typescript Version. With Typescript 5.2.2 Volar works fine, but upgrading to Typescript 5.3.2 broke the auto imports.

You can switch Typescript versions with CTRL+Shift+P -> "Volar: Select Typescript Version". If you don't have 5.2.2 available in the dropdown, you need to install it. In our project it is a pnpm dependency and all we had to do was change the version in the package.json and run `pnpm i`. 

Hope that helps :)

EDIT: Typescript 5.3.3 also breaks the intellisense.

**Reply by ewilan-riviere** [reply]
Same here, I rollback Typescript to `5.2.2` and it works again. I have this bug on a Laravel project, so I suppose a problem with Volar and Typescript ?
